new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java

```diff
@@ -0,0 +1,460 @@
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain(imageName);
+
+    // map to save HTTP response data
+    httpAttrMap = new HashMap<String, String>();
+    httpAttrMap.put("sessioncreatetime", "(.*)sessioncreatetime>(.*)</sessioncreatetime(.*)");
+    httpAttrMap.put("sessionid", "(.*)sessionid>(.*)</sessionid(.*)");
+    httpAttrMap.put("primary", "(.*)primary>(.*)</primary(.*)");
+    httpAttrMap.put("secondary", "(.*)secondary>(.*)</secondary(.*)");
+    httpAttrMap.put("count", "(.*)countattribute>(.*)</countattribute(.*)");
+  }
+
+  @AfterAll
+  void tearDown() {
+  }
+
+  /**
+   * Get the primary and secondary server name and session create time from the factory method,
+   * getOriginalServerAndSessionInfo. Stop the primary server by changing ServerStartPolicy to NEVER
+   * and patching domain. Send a HTTP request to get http session state (count number), primary server
+   * and session create time. Verify that a new primary server is picked and HTTP session state is migrated.
+   *
+   * @param origPrimaryServerName the primary server name from the factory method
+   *                              getOriginalServerAndSessionInfo when the domain is created
+   * @param origSecondaryServerName the secondary server name from the factory method
+   *                                getOriginalServerAndSessionInfo when the domain is created
+   * @param origPrimaryServerName the session create time from the factory method
+   *                              getOriginalServerAndSessionInfo when the domain is created
+   */
+  @ParameterizedTest
+  @MethodSource("getOriginalServerAndSessionInfo")
+  @DisplayName("Stop the primary server, verify that a new primary server is picked and HTTP session state is migrated")
+  @Slow
+  @MustNotRunInParallel
+  public void testSessionMigration(String origPrimaryServerName,
+                                   String origSecondaryServerName,
+                                   String origSessionCreateTime) {
+    final String sessionCreateTimeAttr = "sessioncreatetime";
+    final String primaryServerAttr = "primary";
+    final String countAttr = "count";
+    final String webServiceGetUrl = SESSMIGR_APP_WAR_NAME + "/?getCounter";
+
+    logger.info("Got the primary server {0}, the secondary server {1} "
+        + "and session create time {2} from the factory method getOriginalServerAndSessionInfo",
+            origPrimaryServerName, origSecondaryServerName, origSessionCreateTime);
+
+    // stop the primary server by changing ServerStartPolicy to NEVER and patching domain
+    logger.info("Shut down the primary server {0}", origPrimaryServerName);
+    shutdownServerUsingServerStartPolicy(origPrimaryServerName);
+
+    // get HTTP response data from web service deployed on the cluster
+    Map<String, String> httpDataInfo =
+        processHttpRequest(domainUid + "-" + origSecondaryServerName,
```

Comment:
is the request sent to cluster url or specific server?

Reply:
Sent to the cluster because the webapp is deployed on the cluster